bundles/core/org.eclipse.smarthome.core.thing.xml.test/src/test/groovy/org/eclipse/smarthome/core/thing/xml/test/SystemWideChannelTypesTest.groovy

```diff
@@ -0,0 +1,128 @@
+    }
+
+    @After
+    void tearDown() {
+        SyntheticBundleInstaller.uninstall(getBundleContext(), SYSTEM_CHANNELS_BUNDLE_NAME)
+        SyntheticBundleInstaller.uninstall(getBundleContext(), SYSTEM_CHANNELS_USER_BUNDLE_NAME)
+    }
+
+    @Test
+    void 'assert that System Channels are loaded and unloaded'() {
+        def bundleContext = getBundleContext()
+        def initialNumberOfThingTypes = thingTypeProvider.getThingTypes(null).size()
+
+        def initialNumberOfChannelTypes = systemChannelTypeProvider.getSystemChannelTypes().size()
+
+        // install test bundle
+        Bundle bundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_BUNDLE_NAME)
+        assertThat bundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes + 1)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes + 1)
+
+        // uninstall test bundle
+        bundle.uninstall();
+        assertThat bundle.state, is(Bundle.UNINSTALLED)
+
+        thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes)
+
+    }
+
+    @Test
+    void 'assert that System Channels are used by other binding'() {
+        def bundleContext = getBundleContext()
+        def initialNumberOfThingTypes = thingTypeProvider.getThingTypes(null).size()
+        def initialNumberOfChannelTypes = systemChannelTypeProvider.getSystemChannelTypes().size()
+
+
+        // install test bundle
+        Bundle sysBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_BUNDLE_NAME)
+        assertThat sysBundle, is(notNullValue())
+
+        Bundle sysUserBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_USER_BUNDLE_NAME)
+        assertThat sysUserBundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null)
+        assertThat thingTypes.size(), is(initialNumberOfThingTypes + 2)
+
+        assertThat systemChannelTypeProvider.getSystemChannelTypes().size(), is(initialNumberOfChannelTypes + 1)
+
+    }
+
+    @Test
+    void 'assert that Thing Types have proper channel definitions'() {
+        def bundleContext = getBundleContext()
+        def initialNumberOfThingTypes = thingTypeProvider.getThingTypes(null).size()
+        def initialNumberOfChannelTypes = systemChannelTypeProvider.getSystemChannelTypes().size()
+
+
+        // install test bundle
+        Bundle sysBundle = SyntheticBundleInstaller.install(bundleContext, SYSTEM_CHANNELS_BUNDLE_NAME)
+        assertThat sysBundle, is(notNullValue())
+
+        def thingTypes = thingTypeProvider.getThingTypes(null).findAll{it.getUID().getId().equals("wireless-router") }
+
+        for(ThingType type : thingTypes) {
```

Comment:
If for some reason the list of thing types is empty, the test will pass, but I think it should fail. Better use thingTypes.find() for example and make explicit checks, which are not within a loop